File: keeper/Models/Vault.cs
namespace keeper.Models
{
  public class Vault : IDbItem<int>, IHasCreator
  {
    public int Id { get; set; }
    public string CreatorId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Img { get; set; }
    public bool IsPrivate { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // NOTE Virtual
    public Profile Creator { get; set; }
  }
}

File: keeper.client/src/models/Vault.js
import { Profile } from "./Profile.js";

export class Vault {
  constructor(data) {
    this.id = data.id;
    this.creatorId = data.creatorId;
    this.name = data.name;
    this.description = data.description;
    this.img = data.img;
    this.isPrivate = data.isPrivate;
    this.createdAt = data.createdAt;
    this.updatedAt = data.updatedAt;
    this.creator = new Profile(data.creator);
  }
}
